backend/infrahub/trigger/tasks.py

```diff
@@ -1,22 +1,27 @@
 from typing import TYPE_CHECKING
 
-from infrahub_sdk.utils import compare_lists
 from prefect import get_run_logger, task
 from prefect.automations import AutomationCore
 from prefect.cache_policies import NONE
 from prefect.client.orchestration import PrefectClient
 from prefect.client.schemas.filters import DeploymentFilter, DeploymentFilterName
 
-from .catalogue import triggers
-from .constants import DEPRECATED_STATIC_TRIGGER_NAMES
+from infrahub.trigger.models import TriggerDefinition
+
+# from .catalogue import triggers
 from .models import TriggerType
 
 if TYPE_CHECKING:
     from uuid import UUID
 
 
-@task(name="trigger-setup", task_run_name="Setup triggers in task-manager", cache_policy=NONE)  # type: ignore[arg-type]
-async def setup_triggers(client: PrefectClient) -> None:
+@task(name="trigger-setup", task_run_name="Setup triggers of type {trigger_type.value}", cache_policy=NONE)  # type: ignore[arg-type]
+async def setup_triggers(
+    client: PrefectClient,
+    triggers: list[TriggerDefinition],
+    trigger_type: TriggerType = TriggerType.BUILTIN,
```

Comment:
As we are sending in a list of TriggerDefinition I don't know if we want to have `trigger_type` as another argument. Looks like we have the `.type` attribute on children of TriggerDefinition. Could we instead have a TriggerType.UNDEFINED or even BUILTIN on the parent class?

Reply:
This function is meant to configure all triggers of a given Kind at once, this is the only way to know for sure which one we can delete or not so I think it's best to explicitly define the kind at the top level.
Thinking more about it, we should add a check to ensure that only Trigger of this specific kind have been provided. it will be even clearer